### PR TITLE
[SYCL] Enable SPV_INTEL_runtime_aligned for FPGA only

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
Created to fix pre-commit issues from https://github.com/intel/llvm/pull/5111 (this PR added here as well).

**Comment by MrSidims** [comment]
Looks like pre-commit is passing with the clang FE patch (CUDA failure is a known issue). Reverting back to only Driver changes.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -9041,17 +9041,12 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
...
-        ",+SPV_INTEL_runtime_aligned";
+        ",+SPV_INTEL_arithmetic_fence";
     ExtArg = ExtArg + DefaultExtArg + INTELExtArg;
     if (!C.getDriver().isFPGAEmulationMode())
-      // Enable SPV_INTEL_usm_storage_classes only for FPGA hardware,
-      // since it adds new storage classes that represent global_device and
-      // global_host address spaces, which are not supported for all
-      // targets. With the extension disabled the storage classes will be
-      // lowered to CrossWorkgroup storage class that is mapped to just
-      // global address space.
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:9053]: So is this information no longer relevant/no longer useful to the reader?

Comment (MrSidims) [reply clang/lib/Driver/ToolChains/Clang.cpp:9053]: It's still relevant, but I don't think it's (was) useful. Also note, that there might/will be more exclusive FPGA extensions.

**Review by AGindinson — APPROVED** [review]

**Pull request merged**